UserInterface/GameManager/GameManager.cs

```diff
@@ -69,43 +90,85 @@ private void RunGame()
                 Console.CursorVisible = false;
                 Pair? currentPair = _pairManager.Pair;
```

Comment:
Is logic concerning currentPair  viable? This code looks low level. If it is needed consider to move it to some other class.